### PR TITLE
[MODULAR] Ghostrole Cryo Announcements!

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
So, this adds in one new announcement, and slightly changes up how cryo announcements work in general:

- The cryo computer now has it's own announcement radio, which is *important* if we want ghost role announcements!
- The cryo computer can be configured with custom headsets and to use a custom channel!
- Tarkon, DS2 and Interdyne ghost roles now have cryo ***wake up*** and sleep announcements!
- `announcement_system.dm` is no longer touched by any changes as a result!
- Ghost roles also now get general records for use in cryo announcements- I didn't find any player facing uses for general records, so I think this is safe!
  
  
  
  
  

## How This Contributes To The Skyrat Roleplay Experience
Knowing when your fellow Tarkon/Syndie crewmembers wake up and go to sleep is really nice!

## Changelog

:cl:
add: Port Tarkon, Interdyne and DS-2 now have cryo wake up and sleep announcements.
code: Cryo announcements no longer use the Automated Announcement System.
code: Ghost roles that cryo will have their ghost role name in the cryo message.
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
Oh boy, here we go, github's image syntax isn't what it's documented to be!

Update: Turns out I have to use img tags inside of other html tags- Whatever, those images are really small, so!

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 1]
```diff
@@ -17,24 +17,27 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
+	var/obj/item/radio/headset/radio = /obj/item/radio/headset/silicon/ai
+	var/announcement_channel = RADIO_CHANNEL_COMMON
 
 
 /obj/machinery/computer/cryopod/Initialize(mapload)
 	. = ..()
 	GLOB.cryopod_computers += src
+	radio = new radio(src)
 
 /obj/machinery/computer/cryopod/Destroy()
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm:43]: make sure to del and null the headset in Destroy()

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 2]
```diff
@@ -17,24 +17,27 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
 	/// Used for logging people entering cryosleep and important items they are carrying.
 	var/list/frozen_crew = list()
 	/// The items currently stored in the cryopod control panel.
 	var/list/frozen_item = list()
 
-	var/storage_type = "crewmembers"
-	var/storage_name = "Cryogenic Oversight Control"
+	/// This is what the announcement system uses. No longer uses the AAS because modularity.
+	var/obj/item/radio/headset/radio = /obj/item/radio/headset/silicon/ai
+	var/announcement_channel = RADIO_CHANNEL_COMMON
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: code doc this var

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 3]
```diff
@@ -17,24 +17,27 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
 
 
 /obj/machinery/computer/cryopod/Initialize(mapload)
 	. = ..()
 	GLOB.cryopod_computers += src
+	radio = new radio(src)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm:41]:
```suggestion
	radio = new (src)
```
should work

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 4]
```diff
@@ -431,4 +439,26 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 /obj/machinery/cryopod/blob_act()
 	return // Sorta gamey, but we don't really want these to be destroyed.
 
+// Wake-up notifications
+
+/obj/effect/mob_spawn/ghost_role
+	// For figuring out where the local cryopod computer is. Must be set for cryo computer announcements.
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]:
```suggestion
	/// For figuring out where the local cryopod computer is. Must be set for cryo computer announcements.
```

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 5]
```diff
@@ -431,4 +439,26 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
+
+/obj/effect/mob_spawn/ghost_role/proc/find_control_computer()
+	if(computer_area)
+		for(var/cryo_console as anything in GLOB.cryopod_computers)
+			var/obj/machinery/computer/cryopod/console = cryo_console
+			var/area/area = get_area(cryo_console) // Define moment
+			if(area.type == computer_area)
+				return console
+
+	return null
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]:
```suggestion
/obj/effect/mob_spawn/ghost_role/proc/find_control_computer()
	if(!computer_area)
	    return
	for(var/cryo_console as anything in GLOB.cryopod_computers)
		var/obj/machinery/computer/cryopod/console = cryo_console
		var/area/area = get_area(cryo_console) // Define moment
		if(area.type == computer_area)
			return console
```

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 6]
```diff
@@ -431,4 +439,26 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
+	var/obj/machinery/computer/cryopod/control_computer = find_control_computer()
+	if(control_computer)
+		control_computer.announce("CRYO_JOIN", spawned_mob.real_name, null)
+
+/obj/effect/mob_spawn/ghost_role/proc/find_control_computer()
+	if(computer_area)
+		for(var/cryo_console as anything in GLOB.cryopod_computers)
+			var/obj/machinery/computer/cryopod/console = cryo_console
+			var/area/area = get_area(cryo_console) // Define moment
+			if(area.type == computer_area)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: why not `istype`?

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/cryosleep/code/cryopod.dm (hunk 1 above)
Comment (RimiNosha) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm:43]: Oh yea, woops-

File: modular_skyrat/modules/cryosleep/code/cryopod.dm (hunk 3 above)
Comment (RimiNosha) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm:41]: Nope, that'll instantiate `/obj/item/radio`, unless the brackets change that!

Comment (Zonespace27) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm:41]: Yeah, good point, am goof

File: modular_skyrat/modules/cryosleep/code/cryopod.dm (hunk 6 above)
Comment (RimiNosha) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm]: Istype does subpaths. I do not want it to match subpaths!

**Comment by Zenitheevee** [comment]
Sorry for the Merge conflict.
Anyways, It _is_ worth keeping in mind that the cryo-consoles have Access requirements to draw items out.
Would you mind putting them within the modularized consoles so they're no longer needed to be Var-Edited?
For interdyne it might not be as big of a deal, But in Tarkon its in more as an anti-griefing measure.

**Comment by RimiNosha** [comment]
I thought I did that- fixed!

**Comment by RimiNosha** [comment]
I'm also half-tempted to make cryo computers not require power at all- it'd make the start of Tarkon, when folk are spawning in and potentially cryoing out the most much easier to track. 

Also good if the main station runs out of power, cause it's common for folk to have to go pretty urgently and not have time to spend waiting on a shuttle that may or may not come due to a depowered console-

Which now that I think about it, might be my next PR, cause it glows in the dark, but becomes depowered along with the station despite still glowing- one of the two have to change, whether it doesn't require power, or it requires power and doesn't glow when depowered-

Edit: 👍 for not requiring power for cryo consoles, 👎 for requiring power, for easy opinion measurement!

**Comment by Zenitheevee** [comment]
> I'm also half-tempted to make cryo computers not require power at all- it'd make the start of Tarkon, when folk are spawning in and potentially cryoing out the most much easier to track.
> 
> Also good if the main station runs out of power, cause it's common for folk to have to go pretty urgently and not have time to spend waiting on a shuttle that may or may not come due to a depowered console-
> 
> Which now that I think about it, might be my next PR, cause it glows in the dark, but becomes depowered along with the station despite still glowing- one of the two have to change, whether it doesn't require power, or it requires power and doesn't glow when depowered-
> 
> Edit: 👍 for not requiring power for cryo consoles, 👎 for requiring power, for easy opinion measurement!

Generally speaking, The start of tarkon is a bit hard and thats an understandable thing, Especially when its required items like, i dunno, The modsuits, all guns and the only space suit thats guaranteed before xenos.
While i dont think there *has* been any really bad cases of griefing, One step forward is better than punishing someone because we're one step back.

**Review by Jolly-66 — APPROVED** [review]
Map edits are simple and simply QoL which this PR adds.

File: code/modules/mob_spawn/ghost_roles/mining_roles.dm [hunk 7]
```diff
@@ -287,6 +287,7 @@
 	loadout_enabled = TRUE // SKYRAT EDIT ADDITION - ghost role loadouts
 	quirks_enabled = TRUE // SKYRAT EDIT ADDITION - ghost role loadouts
 	random_appearance = FALSE // SKYRAT EDIT ADDITION
+	computer_area = /area/ruin/syndicate_lava_base/dormitories // SKYRAT EDIT ADDITION - Announce cryo wake-ups!
```
Comment (jjpark-kb) [inline code/modules/mob_spawn/ghost_roles/mining_roles.dm]: needs to be modular.

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 8]
```diff
@@ -110,6 +115,12 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 		else
 			CRASH("Illegal action for ui_act: '[action]'")
 
+/obj/machinery/computer/cryopod/proc/announce(message_type, user, rank)
+	if(message_type == "CRYO_JOIN")
+		radio.talk_into(src, "[user][rank ? ", [rank]" : ""] has woken up from cryo storage.", announcement_channel)
+	if(message_type == "CRYO_LEAVE")
```
Comment (jjpark-kb) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: switch statement here.

**Pull request merged**